Bugfix/fix wallet address types test

gitcoindev opened the pull request:
### Description
This pull request brings bitcoin core changes with taproot bech32m wallet addresses support and fixes wallet_address_types functional test.

### Notes
```
$ test/functional/test_runner.py test/functional/wallet_address_types.py
Temporary test directory at /tmp/test_runner_20221108_195554
WARNING! The following scripts are not being run: ['rpc_estimatefee.py', 'wallet_bumpfee_totalfee_deprecation.py']. Check the test lists in test_runner.py.
Running Unit Tests for Test Framework Modules
..........
----------------------------------------------------------------------
Ran 10 tests in 0.836s

OK
Remaining jobs: [wallet_address_types.py --legacy-wallet, wallet_address_types.py --descriptors]
1/2 - wallet_address_types.py --descriptors passed, Duration: 18 s
Remaining jobs: [wallet_address_types.py --legacy-wallet]
2/2 - wallet_address_types.py --legacy-wallet passed, Duration: 32 s

TEST                                    | STATUS    | DURATION

wallet_address_types.py --descriptors   | ✓ Passed  | 18 s
wallet_address_types.py --legacy-wallet | ✓ Passed  | 32 s

ALL                                     | ✓ Passed  | 50 s (accumulated) 
Runtime: 32 s
```

achow101 committed:
Create a tr() descriptor bech32m DescriptorScriptPubKeyMan by default

achow101 committed:
Mention bech32m in -addresstype and -changetype help

achow101 committed:
Extract Taproot internal keyid with GetKeyFromDestination

achow101 committed:
Store pubkeys in TRDescriptor::MakeScripts

When expanding the scripts for a TRDescriptor, also store the pubkeys in
the FlatSigningProvider.

madnadyka commented on test/functional/tool_wallet.py:
```diff
@@ -69,8 +69,8 @@ def log_wallet_timestamp_comparison(self, old, new):
         wallet_name = self.default_wallet_name if name == "" else name
-        output_types = 3  # p2pkh, p2sh, segwit
         if self.options.descriptors:
+            output_types = 4  # p2pkh, p2sh, segwit, bech32m
```
Please change further keypool size values from 6 to 8 to pass test successfully: https://github.com/BitgesellOfficial/bitgesell/blob/3e51f97438f1868db09d052f9eeca4945493213d/test/functional/tool_wallet.py#L229
https://github.com/BitgesellOfficial/bitgesell/blob/3e51f97438f1868db09d052f9eeca4945493213d/test/functional/tool_wallet.py#L286

gitcoindev replied:
Thank you @madnadyka . I rebased against latest master, rebuilt, retested and pushed the change:

```test/functional/test_runner.py test/functional/wallet_address_types.py
Temporary test directory at /tmp/test_runner_20221122_104012
WARNING! The following scripts are not being run: ['rpc_estimatefee.py', 'wallet_bumpfee_totalfee_deprecation.py']. Check the test lists in test_runner.py.
Running Unit Tests for Test Framework Modules
..........
----------------------------------------------------------------------
Ran 10 tests in 0.521s

OK
Remaining jobs: [wallet_address_types.py --legacy-wallet, wallet_address_types.py --descriptors]
1/2 - wallet_address_types.py --descriptors passed, Duration: 14 s
Remaining jobs: [wallet_address_types.py --legacy-wallet]
2/2 - wallet_address_types.py --legacy-wallet passed, Duration: 28 s

TEST                                    | STATUS    | DURATION

wallet_address_types.py --descriptors   | ✓ Passed  | 14 s
wallet_address_types.py --legacy-wallet | ✓ Passed  | 28 s

ALL                                     | ✓ Passed  | 42 s (accumulated) 
Runtime: 28 s
```

gitcoindev committed:
test/functional/tool_wallet: Increase pool size to 8.

Increase the keypool size value from 6 to 8 to accomodate adding bech32m
address support and pass the wallet_address_types test case.
